### PR TITLE
I10 create category

### PR DESCRIPTION
**Pull request by gabegaghi**
## Description
Added functionality of creating a new category

## Related Issue
- [x] This pull request relates to at least one particular issue

  
  - [Create Category](https://github.com/ps-toronto-team-4/.github/issues/10)

## Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

None.

### CODE REVIEW

**Comment by HKanwal** [comment]
@gabegaghi please resolve merge conflict first. Changing this into a draft until those are addressed.

**Review by HKanwal — CHANGES_REQUESTED** [review]
See requested changes. Also, two bugs were noticed:
1. The alert 'This category is already taken' shows even if it is a new category name and the submit succeeds.
2. User is able to create category with a color that is already taken. If this is allowed, remove the alert that shows 'Color already taken'

File: components/Themed.tsx [hunk 1]
```diff
@@ -43,3 +45,12 @@ export function View(props: ViewProps) {
 
   return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
 }
+
+export function RequiredField({ check, input }: { check: boolean, input: string }) {
```
[the remark this answers is not in the document]
Comment (gabegaghi) [reply components/Themed.tsx:49]: but it's 'check?' as in 'true: yes, check if the field is filled out' vs 'false: no need to check if it is filled out yet'. so what would a better name be?

File: screens/CreateCategoryScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { useMutation, useLazyQuery } from '@apollo/client';
+import { CreateCategoryDocument, CreateCategoryMutation, GetCategoriesDocument, GetCategoriesQuery } from '../components/generated';
+import { getAuth } from '../components/GetAuth';
+
+export default function CreateCategoryScreen({ navigation }: RootStackScreenProps<'CreateCategory'>) {
+
+  const [name, setName] = useState("")
+  const [details, setDetails] = useState("")
+  const [color, setColor] = useState("")
+  const [check, setCheck] = useState(false)
```
[the remark this answers is not in the document]
Comment (gabegaghi) [reply screens/CreateCategoryScreen.tsx:20]: added a comment

File: components/Themed.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (HKanwal) [reply components/Themed.tsx:49]: Got it. Keeping it as check is fine.

**Pull request merged**